lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1587,7 +1600,7 @@ def main():
         instance_type=dict(default='t2.micro', type='str'),
         user_data=dict(type='str'),
         tower_callback=dict(type='dict'),
-        ebs_optimized=dict(type='bool'),
+        ebs_optimized=dict(type='raw'),
```

Comment:
If someone passes something other than a boolean or 'auto', `boolean()` will cause a TypeError. Using `choices` with a list containing 'auto' and all the possible booleans for choices is one solution, or you could use https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/validation.py#L436 and if the value isn't 'auto' or a valid boolean fail at the beginning of the module for invalid type.

LGTM besides that. The hardcoding is unfortunate but seems like the best choice for now.